### PR TITLE
Features/HDB 56 rnbw not displaying vesting page

### PR DESCRIPTION
**Pull request by bill-ov**
… while on Vesting page

## Description of Changes

Fix not updating RNBW and xRNBW balance on Vesting page when logged in metamask account while on Vesting page

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDB-56)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDEV-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/sushi-hooks/queries/useTokenBalance.ts [hunk 1]
```diff
@@ -36,6 +36,10 @@ const useTokenBalance = (tokenAddress?: string, accountAddress?: string) => {
     }
   }
 
+  useEffect(() => {
+    setTargetAddress(account)
```
Comment (schystz) [inline src/sushi-hooks/queries/useTokenBalance.ts]: I think we still need this condition here (see l20) when assigning the targetAddress. We need to make it backwards compatible for others using `useTokenBalance()` hook which explicitly provides an `accountAddress` param

```suggestion
    setTargetAddress(accountAddress ?? account)
```

**Review by schystz — COMMENTED** [review]
Tested this on preview site, fix looks good! Just 1 comment to maintain backwards compatibility.

**Review by tracyarciaga — APPROVED** [review]
Approve this on my end, code looks fine and tested working, but pls resolve @schystz 's comments first.

File: src/sushi-hooks/queries/useTokenBalance.ts (hunk 1 above)
Comment (bill-ov) [reply src/sushi-hooks/queries/useTokenBalance.ts]: @schystz done resolving your feedback

File: src/sushi-hooks/queries/useTokenBalance.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export interface BalanceProps {
 const useTokenBalance = (tokenAddress?: string, accountAddress?: string) => {
   const [balance, setBalance] = useState<BalanceProps>({ value: BigNumber.from(0), decimals: 18 })
   const { account } = useActiveWeb3React()
-  const [targetAddress] = useState(accountAddress ?? account)
+  const [targetAddress, setTargetAddress] = useState(account)
```
Comment (schystz) [inline src/sushi-hooks/queries/useTokenBalance.ts]: @bill-ov I only mentioned this line to use as basis for my comment on line 40. Let's keep the original logic here pls

```suggestion
  const [targetAddress, setTargetAddress] = useState(accountAddress ?? account)
```

Comment (bill-ov) [reply src/sushi-hooks/queries/useTokenBalance.ts]: Okay James. I'll change it after lunch. Sorry for not noticing your comment early

Comment (bill-ov) [reply src/sushi-hooks/queries/useTokenBalance.ts]: Done changing it James

Comment (schystz) [reply src/sushi-hooks/queries/useTokenBalance.ts]: Thanks @bill-ov ! However the linters & unit tests seems to be failing now. Can you pls take care of this? No need to do it now, but will leave it up to you. Not a priority

**Review by 0xAplki — CHANGES_REQUESTED** [review]
Please check error in github actions. there seemed to be an error in installing dependencies. it's related to the yarn lock committed here

File: src/sushi-hooks/queries/useTokenBalance.ts (hunk 2 above)
Comment (bill-ov) [reply src/sushi-hooks/queries/useTokenBalance.ts]: Hi @schystz. Done resolving the issue. All tests are now passed

**Review by schystz — APPROVED** [review]

**Pull request merged**